x/perp/v2/keeper/dnr.go

```diff
@@ -0,0 +1,90 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	"github.com/NibiruChain/collections"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// DnRGCFrequency is the frequency at which the DnR garbage collector runs.
+const DnRGCFrequency = 1000
+
+// IntValueEncoder instructs collections on how to encode a math.Int.
+// TODO: move to collections.
+var IntValueEncoder collections.ValueEncoder[math.Int] = intValueEncoder{}
+
+type intValueEncoder struct{}
+
+func (i intValueEncoder) Encode(value math.Int) []byte {
+	v, err := value.Marshal()
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
+func (i intValueEncoder) Decode(b []byte) math.Int {
+	var v math.Int
+	err := v.Unmarshal(b)
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
+func (i intValueEncoder) Stringify(value math.Int) string {
+	return value.String()
+}
+
+func (i intValueEncoder) Name() string {
+	return "math.Int"
+}
+
+// IncreaseTraderVolume adds the volume to the user's volume for the current epoch.
+func (k Keeper) IncreaseTraderVolume(ctx sdk.Context, user sdk.AccAddress, volume math.Int) {
+	currentEpoch, err := k.DnREpoch.Get(ctx)
```

Comment:
refactor suggestion: I would pass in `currentEpoch` as an argument and move the getter to be called just before this function for two reasons:
1. You'll need the `currentEpoch` value to compute the taker fee on this part of clearing_house.go:
```go
	// update user volume
	k.IncreaseTraderVolume(ctx, traderAddr, positionResp.ExchangedNotionalValue.Abs().TruncateInt())

	transferredFee, err := k.transferFee(ctx, market.Pair, traderAddr, positionResp.ExchangedNotionalValue,
		market.ExchangeFeeRatio, market.EcosystemFundFeeRatio,
	)
```
Here (in a future PR), you'll need the `currentEpoch` value to get the trader's DnR table

2. When you run this in a loop for garbage collection, you can just do 1 state read for the epoch and re-use the value while iterating through the trader addresses to make it a bit more efficient